### PR TITLE
Only show skips in summaries

### LINKED ISSUE #1801: Tweak skip output
Make `testthat.progress.verbose_skips = FALSE` the default, and add additional info to the summary bullets.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1801

### CODE REVIEW

File: tests/testthat/_snaps/reporter-progress.md [hunk 1]
```diff
@@ -315,19 +315,11 @@
     - |     1   0 | reporters/skips                                                 
     \ |     2   0 | reporters/skips                                                 
     v |     2   0 | reporters/skips
-    --------------------------------------------------------------------------------
-    Skip ('reporters/skips.R:2:3'): regular skip
-    Reason: regular skip
-    
-    Skip ('reporters/skips.R:6:3'): skip with details
-    Reason: longer skip:
-    this is what happened
```
Comment (DavisVaughan) [inline tests/testthat/_snaps/reporter-progress.md:324]: I guess we don't see details anymore, but that's probably ok. Details are probably more useful for interactive use when you are actually looking at the test in question when debugging something (i.e. you are reminding yourself why you skip this test)

File: R/reporter-progress.R [hunk 2]
```diff
@@ -528,16 +518,36 @@ strpad <- function(x, width = cli::console_width()) {
...
-  tbl <- table(skips)
-  paste0(cli::symbol$bullet, " ", names(tbl), " (", tbl, ")")
+  locs <- map_chr(skips, expectation_location)
+  locs_by_skip <- split(locs, message)
+  n <- lengths(locs_by_skip)
+  skip_summary <- map_chr(locs_by_skip, paste, collapse = ", ")
+
+  bullets <- paste0(
+    cli::symbol$bullet, " ", names(locs_by_skip), " (", n, "): ", skip_summary
+  )
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/reporter-progress.R:545]: It's still a heck of a lot less room than 2 lines per skip 😄

File: R/reporter-progress.R [hunk 3]
```diff
@@ -528,16 +518,36 @@ strpad <- function(x, width = cli::console_width()) {
...
-  skips <- gsub("Reason: ", "", skips)
-  skips <- gsub(":?\n(\n|.)+", "", skips) # only show first line
+  message <- map_chr(skips, "[[", "message")
+  message <- gsub("Reason: ", "", message)
+  message <- gsub(":?\n(\n|.)+", "", message) # only show first line
 
-  tbl <- table(skips)
-  paste0(cli::symbol$bullet, " ", names(tbl), " (", tbl, ")")
+  locs <- map_chr(skips, expectation_location)
+  locs_by_skip <- split(locs, message)
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/reporter-progress.R:539]: Yeah, I'm not sure why this code even exists, TBH.

**Review by jennybc — APPROVED** [review]
I think this is a good move.

**Pull request merged**